pkg/dashboard/ui/src/app/shared/services/import.service.js

```diff
@@ -74,28 +96,109 @@
+                        return NuclioFunctionsDataService.createFunction(func, projectName)
+                            .catch(function (error) {
+                                if (error.status === 409) {
+                                    if (lodash.has(conflictProjectsData, projectName)) {
+                                        return conflictProjectsData[projectName].push(func);
+                                    } else {
+                                        return lodash.set(conflictProjectsData, [projectName], [func]);
+                                    }
+                                }
+                            });
                     });
 
-                    if (lodash.isFunction(onSuccess)) {
-                        onSuccess()
-                    }
+                    return $q.all(checkedFunctionList);
                 })
                 .catch(function (error) {
                     DialogsService.alert($i18next.t('common:ERROR_MSG.IMPORT_PROJECT', {lng: lng}))
                         .then(function () {
                             if (lodash.isFunction(onFailure)) {
-                                onFailure(error)
+                                onFailure(error);
                             }
                         });
                 });
         }
+
+        /**
+         * Opens confirm dialog
+         * @returns {Object} closeOptions
+         */
+        function openConfirmDialog(functionName, projectName) {
+            var data = {
+                displayAllOptions: displayAllOptions,
+                dialogTitle: $i18next.t('common:OVERRIDE_FUNCTION_CONFIRM', {lng: lng, functionName: functionName, projectName: projectName})
+            };
+
+            return ngDialog.open({
+                template: '<igz-import-project-dialog ' +
+                    'data-display-all-options="ngDialogData.displayAllOptions" ' +
+                    'data-close-dialog="closeThisDialog({action, option})" ' +
+                    'data-dialog-title="ngDialogData.dialogTitle">' +
+                    '</igz-import-project-dialog>',
+                plain: true,
+                data: data,
+                className: 'ngdialog-theme-iguazio image-dialog'
+            })
+                .closePromise
+                .then(function (closeOptions) {
+                    return closeOptions;
+                });
+        }
+
+        /**
+         * Handles action and option for resolve conflict function
+         */
+        function resolveConflict() {
+            var currentProject = lodash.keys(conflictProjectsData)[0];
+            var currentFunction = lodash.get(conflictProjectsData, [currentProject] + '[0]');
+
+            openConfirmDialog(currentFunction.metadata.name, currentProject)
+                .then(function (data) {
+                    if (data.value.option === 'allProjects') {
+                        lodash.mapKeys(conflictProjectsData, function (functionsList, projectName) {
```

Comment:
No reason to use `lodash.mapKeys` only to iterate the object's items.

`lodash.mapKeys` is used to create a shallow copy of an object's properties, where the keys go through some transformation (and values remain the same).

For example:

```js
var obj = {
    abc: '123',
    def: '456',
    egi: '789'
};

// note that the return value of `lodash.mapKeys` is not being ignored
var newObj = lodash.mapKeys(obj, function (value, key) {
    // note that the iteratee function returns a value, not just undefined
    // this value will be the key of a property in the newly created object
    return key.charAt(0) + value.chartAt(0);
});

console.log(JSON.stringify(newObj));
// => {a1:'123',d4:'def',e7:'789'} note the values stay the same
```
- Note that you must return something in the iteratee function to create the new keys. Otherwise, the keys will not be mapped and the specific choice in `lodash.mapKeys` is useless.
- Note that the new created object returned from `lodash.mapKeys` is being used (in this case, stored in a variable), otherwise _the newly created object is discarded_ and the specific choice in `lodash.mapKeys` is useless. Other ways to using the returned value other than assigning it to a variable, is to send it as a function argument in a function call, or return it as a return value from a function, or just being a part of some expression.

All mapping methods have similar notes. `_.map` and `_.mapValues` and `Array.prototype.map`, etc.

If you just want to iterate the values and/or keys of a collection (an array or an object) to perform some action with each of them, you can simply use `lodash.forEach` (there are also `lodash.forIn` and `lodash.forOwn` for objects only, not arrays, but `forEach` works for objects as well, and there's no reason not to use it).

`lodash.forEach` _does_ have a return value too, but it's just the same collection you provide as the first parameter, unmodified, so ignoring its return value _does_ make sense (unlike ignoring the return value of a mapping function).